DESIGN.md

```diff
@@ -118,6 +118,36 @@ bcd.addTest('css.properties.custom-property', "(function() {return CSS.supports(
 
 Tips: make sure that all return statements will return a boolean, and implement thorough feature checking.
 
+#### Importing code from other tests
+
+Sometimes, some features will depend on the setup and configuration from other features, especially with APIs.  To prevent repeating the same code over and over again, you can import code from other custom tests to build new ones quicker.  The syntax to specify a test import is the following: `<%ident:varname%>`, where `ident` is the full identifier to import from, and `varname` is what to rename the `instance` variable from that test to.
+
+Example:
+
+The following JSON...
+
+```
+{
+  "api": {
+    "AudioContext": {
+      "__base": "var instance = new (window.AudioContext || window.webkitAudioContext)();"
+    },
+    "AudioDestinationNode": {
+      "__base": "<%api.AudioContext:audioCtx%> if (!audioCtx) {return false}; var instance = audioCtx.destination;"
```

Comment:
This is just an example, but the `if (!audioCtx) {return false}` makes me nervous about trusting the results just like the issue with assuming interface objects exist. There might be many ways to get an instance of this, and not finding one isn't very conclusive evidence that the subfeature isn't supported. Also, wouldn't the code throw an exception more often than not?

Just assuming that `audioCtx` was set and letting the code fail if not would be preferable, I think.

Reply:
We can certainly do that, though we should implement, in update-bcd, a piece of code to check to see if `AudioContext` is supported and compare it to the `null` values.  I've mentioned this in #506!